src/main/java/io/vertx/ext/sql/SQLRowStream.java

```diff
@@ -0,0 +1,33 @@
+package io.vertx.ext.sql;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.streams.ReadStream;
+
+/**
+ * A ReadStream of Rows from the underlying RDBMS. This class follows the ReadStream semantics and will automatically
+ * close the underlying resources if all returned rows are returned. For cases where the results are ignored before the
+ * full processing of the returned rows is complete the close method **MUST** be called in order to release underlying
+ * resources.
+ *
+ * The interface is minimal in order to support all SQL clients not just JDBC.
+ */
+@VertxGen
+public interface SQLRowStream extends ReadStream<JsonArray> {
+
+    /**
+     * Will convert the column name to the json array index
+     *
+     * @param name
+     * @return
+     */
+    int column(String name);
```

Comment:
Well, giving twice or more time the same column in a row from the database seems a little bit weird ? I know it's possible to execute this kind of query in a database : 
```sql
select ap_app, ap_app, ap_app from s_application;
```
```
AP_APP           AP_APP           AP_APP
---------------- ---------------- ----------------
test            test            test
```
This is redundant, and space consuming (depending of datatype ofc). However,  with a bit of a step back, we can design a Pump stream adapter class that convert the JsonArray to a JSonObject ... may be a good solution.

About mutator : I'm sorry, I wouldn't saying a mutator for controlling the number of row pushed in the stream, but to allow user to manage the cursor size of the underlying RowSet implentation.